src/js/libcs/OpDrawing.js

```diff
@@ -106,6 +106,183 @@ evaluator.fillcircle$2 = function(args, modifs) {
+        } else {
+            var num = 500; // Number of segments
+
+            //  mod 2 pi in case startAngle > endAngle
+            if (startAngle > endAngle) endAngle = endAngle + Math.PI * 2;
+
+            // divide segments --  rotate counterclockwise if necessary
+            var ntler = !cclock ? (endAngle - startAngle) / num : -(2 * Math.PI - endAngle + startAngle) / num;
+
+            // drawing
+            csctx.moveTo(rad * Math.cos(startAngle), rad * Math.sin(startAngle));
+            var angl;
+            for (var ii = 0; ii <= num; ii++) {
+                angl = startAngle + ii * ntler;
+                csctx.lineTo(rad * Math.cos(angl), rad * Math.sin(angl));
+            }
+        }
+
+
+        if (df === "F" || false) {
+            csctx.fillStyle = Render2D.lineColor;
+            csctx.closePath();
+            csctx.fill();
+        }
+
+        if (df === "D") {
+            csctx.stroke();
+        }
+        csctx.restore();
+
+    } else { // segment case
+        var ptA = eval_helper.extractPoint(v1);
+        var ptB = eval_helper.extractPoint(v2);
+        var ptC = eval_helper.extractPoint(v3);
+
+        // dists
+        var dAB = (ptA.x - ptB.x) * (ptA.x - ptB.x) + (ptA.y - ptB.y) * (ptA.y - ptB.y);
+        var dAC = (ptA.x - ptC.x) * (ptA.x - ptC.x) + (ptA.y - ptC.y) * (ptA.y - ptC.y);
+        var dBC = (ptC.x - ptB.x) * (ptC.x - ptB.x) + (ptC.y - ptB.y) * (ptC.y - ptB.y);
+
+        // if 2 points are the same return nada;
+        if (dAB < 1e-12 || dAB < 1e-12 || dAB < 1e-12) return nada;
+
+        // drawing points
+        var pt1, pt2, pt3;
+
+        // check by dets if B is in the middle
+        var randP = List.realVector([10 * Math.random(), 10 * Math.random(), 10 * Math.random()]);
+        var detABP = List.det3(a, b, randP).value.real;
+        var detACP = List.det3(b, c, randP).value.real;
+        var Bmiddle = detABP * detACP > 0;
+
+        // handle modifs
+        if (modifs !== null) {
+            // workaround since conic width !== line width in default settings
+            if (modifs.size == null) modifs.size = CSNumber.real(4 / 2.5);
+
+            Render2D.handleModifs(modifs, Render2D.conicModifs);
+        }
+
+
+        // if B is in the middle we are fine
+        if (Bmiddle) {
+            pt1 = ptA;
+            pt2 = ptC;
+            Render2D.drawsegcore(pt1, pt2);
+        } else { // nasty case -- B not in the middle -- we have 2 ray to infinity
+
+            // flip the orientation to the right side 
+            var sflip = dAB > dBC ? 1 : -1;
+
+            // first ray
+            // get direction and normalise
+            var dx = sflip * (ptA.x - ptB.x);
+            var dy = sflip * (ptA.y - ptB.y);
+            var norm = Math.sqrt(dx * dx + dy * dy);
+            dx = dx / norm;
+            dy = dy / norm;
+
+            var ptATmp = eval_helper.extractPoint(v1);
```

Comment:
Why not just reuse `ptA` resp. `ptC` here? Am I missing something, or should these be exactly like this new invocation of `extractPoint`? If, for some reason I don't see, `extractPoint` is really required, you'll again have to check `ptATmp.ok` I guess.